docs/tutorials/plugin-provider.md

```diff
@@ -0,0 +1,44 @@
+# Plugin provider
+
+The "Plugin" provider allows to integrate ExternalDNS with DNS providers via an HTTP interface.
+The Plugin provider implements the Provider interface. Instead of implementing code specific to a provider, it implements an HTTP client that sends request to an HTTP API.
+The idea behind it is that providers can be implemented in a separate program: these programs expose an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars in the same pod of the ExternalDNS container and listen on localhost only. This is not strictly a requirement, but we would not recommend other setups.
+
+## Architectural diagram
+
+![Plugin provider](../img/plugin-provider.png)
+
+## API guarantees
+
+Providers implementing the HTTP API have to keep in sync with changes to the Go types `plan.Changes` and `endpoint.Endpoint`. We do not expect to make significant changes to those types given the maturity of the project, but we can't exclude that changes will need to happen. We commit to publishing changes to those in the release notes, to ensure that providers implementing the API can keep providers up to date quickly.
+
+## Implementation requirements
+
+The following table represents the methods to implement mapped to their HTTP method and route.
+
+| Provider method | HTTP Method | Route |
+| --- | --- | --- |
+| Records | GET | /records |
+| ApplyChanges | POST | /records |
+| PropertyValuesEqual | POST | /propertyvaluesequal |
+| AdjustEndpoints | POST | /adjustendpoints |
+
+Additionally, the server needs to respond to `GET` requests on `/` to negotiate versions by content type as described in [this document](http://opensource.zalando.com/restful-api-guidelines/#114). The server needs to respond to those requests by reading the `Accept` header and responding with a corresponding `Vary` header including the value `Content-Type` and a `Content-Type` header specifying the supported media type format and version.
+
+**NOTE**: all responses sent with a status code different from `20X` will be considered a failure on ExternalDNS' side.
```

Comment:
Maybe we should elaborate on 4xx and 5xx and other errors. What does the client do?
5xx -> should be retried
4xx -> log error or fatal?
3xx -> ?
2yx -> y != 0 ??